pkg/services/ngalert/ngalert.go

```diff
@@ -316,10 +321,28 @@ func subscribeToFolderChanges(logger log.Logger, bus bus.Bus, dbStore api.RuleSt
 	})
 }
 
+// shouldRun determines if AlertNG should init or run anything more than just the migration.
+func (ng *AlertNG) shouldRun() bool {
+	return ng.Cfg.UnifiedAlerting.IsEnabled()
+}
+
 // Run starts the scheduler and Alertmanager.
 func (ng *AlertNG) Run(ctx context.Context) error {
 	ng.Log.Debug("Starting")
 
+	// Migration is called, even if UA is disabled. If UA is disabled though, this will do nothing except set the
+	// instance-wide migration state to false. This is necessary to allow force_migration to revert UA only when moving
+	// from legacy to UA.
+	err := ng.upgradeService.Run(ctx)
+	if err != nil {
+		return err
+	}
+
+	if !ng.shouldRun() {
+		return nil
+	}
+	ng.Log.Debug("Running")
```

Comment:
Let's keep "Starting" here, and add a separate log message when only upgrade service is starting. Something like "Starting (upgrade only)"